nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1069,3 +1082,42 @@ def on_train_batch_end(self, outputs, dataloader_iter: Any, batch_idx: int, unus
                     # Reset the optimizer update skipped to `None` - this is to prevent scheduler no-ops during
                     # accumulated gradient updates.
                     grad_scaler.optimizer_update_skipped = None
+
+    def _reset_activation_checkpointing_args(self):
+        if self.cfg.get('megatron_amp_O2', False):
+            base_module = self.model.module
+        else:
+            base_module = self.model
```

Comment:
I ran into this issue as well but I think we can solve it with this: 
```
    @property
    def model(self):
        if isinstance(self._model, list):
            return [model.module if isinstance(model, Float16Module) else model for model in self._model]
        elif isinstance(self._model, Float16Module):
            return self._model.module
        else:
            return self._model
```
This also handles the interleaved case, which I think you will have to update your code for.

Reply:
@markelsanz14 can you please address Eric's comment so that we can merge this?

Reply:
Yes, sure. @ericharper which class should this go in? MegatronGPTModel or one of its parent classes?

Reply:
MegatronGPTModel